ppl/src/test/java/org/opensearch/sql/ppl/antlr/PPLSyntaxParserTest.java

```diff
@@ -112,5 +112,25 @@ public void can_parse_simple_query_string_relevance_function() {
         "SOURCE=test | WHERE simple_query_string([\"Tags\" ^ 1.5, Title, `Body` 4.2], 'query',"
             + "analyzer=keyword, quote_field_suffix=\".exact\", fuzzy_prefix_length = 4)"));
   }
+
+  @Test
+  public void testDescribeCommandShouldPass() {
+    ParseTree tree = new PPLSyntaxParser().analyzeSyntax("describe t");
+    assertNotEquals(null, tree);
+  }
+
+  @Test
+  public void testDescribeFieldsCommandShouldPass() {
+    ParseTree tree = new PPLSyntaxParser().analyzeSyntax("describe t | fields a,b");
```

Comment:
What is the use case for this example? It seems little odd because if someone wants  a,b as result he would specify a,b instead of describing table and filtering again. 

Fields command is to project menitoned columns from the result set. A plausible usecase could be describe t | fields 2, 3 which implies give me second and third column names.

Also if someone appends other commands to describe, what is the expected behavior. I am assuming we will be calculating on the result set provided by prior describe command.

Reply:
> I am assuming we will be calculating on the result set provided by prior describe command.

That's correct. When appending other commands to describe, the behavior is to query the metadata table, instead of the data table itself (as expected for the pipe syntax).

An example of the usage of `fields` can be seen [here](https://github.com/opensearch-project/sql/pull/646/commits/d4dd52fe60a5420c4ff4497e0ef05026bdfe1ac9)

The fields does not refer to the data table's fields, but the metadata table's fields. [Here](https://github.com/seankao-az/sql/blob/d4dd52fe60a5420c4ff4497e0ef05026bdfe1ac9/integ-test/src/test/java/org/opensearch/sql/ppl/DescribeCommandIT.java#L30-L57) you can see the full list of such fields.

Reply:
> A plausible usecase could be describe t | fields 2, 3 which implies give me second and third column names.

Interestingly, we *could* support that using the following syntax:
```
describe t | where ORDINAL_POSITION=2 or ORDINAL_POSITION=3
```
However, it doesn't quite work yet at the moment, due to [type mismatch](https://github.com/opensearch-project/sql/blob/7cbb121cb3ad144a5e6d5b9acb9ae8e6da01688e/opensearch/src/main/java/org/opensearch/sql/opensearch/storage/system/OpenSearchSystemIndexSchema.java#L39-L64)
```
❯ curl -XPOST https://localhost:9200/_plugins/_ppl -u 'admin:admin' -k -H 'Content-Type: application/json' -d '{"query": "describe opensearch_dashboards_sample_data_flights | where ORDINAL_POSITION=0"}'

{
  "error": {
    "reason": "Invalid Query",
    "details": "= function expected {[BYTE,BYTE],[SHORT,SHORT],[INTEGER,INTEGER],[LONG,LONG],[FLOAT,FLOAT],[DOUBLE,DOUBLE],[STRING,STRING],[BOOLEAN,BOOLEAN],[TIMESTAMP,TIMESTAMP],[DATE,DATE],[TIME,TIME],[DATETIME,DATETIME],[INTERVAL,INTERVAL],[STRUCT,STRUCT],[ARRAY,ARRAY]}, but get [STRING,INTEGER]",
    "type": "ExpressionEvaluationException"
  },
  "status": 400
}%
❯ curl -XPOST https://localhost:9200/_plugins/_ppl -u 'admin:admin' -k -H 'Content-Type: application/json' -d '{"query": "describe opensearch_dashboards_sample_data_flights | where ORDINAL_POSITION=\"0\""}'

{
  "error": {
    "reason": "Invalid Query",
    "details": "invalid to get integerValue from value of type STRING",
    "type": "ExpressionEvaluationException"
  },
  "status": 400
}
```